[depends] update to Boost 1.70

Sjors opened the pull request:
Version [1.70](https://www.boost.org/users/history/version_1_70_0.html) is most recent.

Versions needed for:
* 1.66: #12557: fixes the single arm64 configuration ([06ee5b5](https://github.com/ruslo/hunter/commit/06ee5b54ef310b376cdaa7928f89b4995ceb2efc))

Sjors commented:
Other than iOs support, are there are any other reasons for updating the Boost version? Otherwise this can wait until #12557 is closer to merge readiness.

@fanquake [suggested](
https://github.com/bitcoin/bitcoin/pull/12557#discussion_r259188631) bumping straight to 1.69

MarcoFalke commented:
This is sporadically updated to the latest version, so might as well do 1.69

Sjors commented:
Mmm, Travis build for 32-bit ARM is [unhappy](https://travis-ci.org/bitcoin/bitcoin/jobs/496964756):

`undefined reference to `boost::system::system_category()`

I'll try if 1.69 magically makes that problem go away.

fanquake commented:
Concept ACK. Will review release notes and test depends builds etc.

fanquake commented:
Relevant release notes from Boost` 1.65.0` -> `1.69.0`:

## [1.69.0](https://www.boost.org/users/history/version_1_69_0.html)
### Filesystem
- Fixed crashes in boost::filesystem::copy due to undefined behavior in the implementation.
- Fixed undefined behavior in boost::filesystem::directory_iterator implementation.
- Fixed MSVC warnings about narrowing conversions. 

### System
- Boost.System is now header-only. A stub library is still built for compatibility, but linking to it is no longer necessary.
- The destructor of error_category is now protected and no longer virtual. This is a potentially breaking change, but its impact is expected to be limited.


## [1.68.0](https://www.boost.org/users/history/version_1_68_0.html)

Boost.WinAPI has been updated to target Windows 7 by default, where possible. In previous releases Windows Vista was the default.

### Test
- Breaking change: The master_test_suite_t object is no more copyable

## [1.67.0](https://www.boost.org/users/history/version_1_67_0.html)

### Filesystem
- Fix static initialization issue that caused a crash if path operations were used before main()

### Test
- Breaking change: Adding test cases with the same name to the same test suite is now reported as an error. See the changes log for more details.

## [1.66.0](https://www.boost.org/users/history/version_1_66_0.html)

### Thread
- Many Windows related fixes.

## [1.65.0](https://www.boost.org/users/history/version_1_65_0.html)

### Test
- C++17 compatibility (tested on VS2017)

Breaking Changes
- Some API changes for the loggers
- Deprecating `BOOST_GLOBAL_FIXTURE` in favor of `BOOST_TEST_GLOBAL_FIXTURE` and `BOOST_TEST_GLOBAL_CONFIGURATION`.

promag commented:
Concept ACK.

Then this could be updated:https://github.com/bitcoin/bitcoin/blob/ce6762030f814bc737bb0a097c17201e0b2edbc5/src/wallet/walletutil.cpp#L70-L72

hebasto commented:
Concept ACK.
Agree with @promag's comments.

Sjors commented:
Rebased and adjusted the version in dependencies.md. We can't change `walletutil.cpp` because the minimum Boost version (1.47.0) doesn't change with this PR. Changing the minimum version can be done in a followup, but there's no urgent need afaik.

Sjors committed:
[depends] boost: update to 1.70

Sjors commented:
I switched from Boost 1.69 to 1.70

practicalswift commented:
Concept ACK

Sjors commented:
Very lightly tested the macOS Gitian QT build.

laanwj commented:
ACK 2620e24b83d16bf0f2bfe360dee1e98b4be59ca5, let's do this early in the 0.19 cycle

laanwj commented on build-aux/m4/ax_boost_system.m4:
```diff
@@ -109,7 +108,7 @@ AC_DEFUN([AX_BOOST_SYSTEM],
             fi
             if test "x$ax_lib" = "x"; then
-                AC_MSG_ERROR(Could not find a version of the boost_system library!)
+                AC_MSG_ERROR(Could not find a version of the library!)
```
This reverts (part of) 25333a26197b55c23debb38740ed875999aa795a but I don't know if we care.

Sjors replied:
I didn't realize we were using a patch on top of https://www.gnu.org/software/autoconf-archive/ax_boost_base.html

Sjors replied:
Oops: #17010

MarcoFalke commented:
It seems that not even debian experimental is at 1.70. I guess that means we are the beta testers? Let's hope debian ships it within the next couple of months.

kristapsk commented:
On Gentoo it exists but is masked (not stable).

MarcoFalke commented:
For reference:

https://packages.debian.org/source/sid/boost-defaults
https://apps.fedoraproject.org/packages/boost/overview/
https://packages.ubuntu.com/eoan/libboost-all-dev
https://www.archlinux.org/packages/extra/x86_64/boost/
https://packages.gentoo.org/packages/dev-libs/boost
